tests/apollo/util/bft.py

```diff
@@ -1109,6 +1180,31 @@ async def get_metric(self, replica_id, bft_network, mtype, mname, component='rep
                     else:
                         return value
 
+    async def wait_for_view_with_threshold(self, view_number: int, threshold: Optional[int] = None,
+                                           timeout_seconds: float = 45, sleep_seconds: float = .25, error_msg=None):
+        """
+        Waits for at least threshold replicas to reach view view_number
+        :param view_number: The view number to reach
+        :param threshold: The minimal amount of replicas which are expected to reach view view_number,
+                          Defaults to 2f + 2c + 1
+        :param timeout_seconds: Maximal amount of seconds to wait
+        :param sleep_seconds: Time to sleep between samples, in seconds
+        :param error_msg: Error to display in case of timeout
+        :return: The number of replicas who reached view view_number at the time of the last sample
+        """
+        threshold = threshold if threshold else 2 * self.config.f + 2 * self.config.c + 1
```

Comment:
I think it is enough to have a regular bft quorum (2f + c +1) replicas in the new view.
Indeed, in order to complete the view change, the new primary has to get at least 2f + 2c +1 viewchange messages from different replicas, but it is enough to consider a view as valid view if the replicas are able to commit in it, and for that, it is enough to have only 2f + c + 1 replicas.
I suggest to also ask @guyg7

Reply:
I think that this change is out of scope (affects too many cases)